lib/symboldatabase.cpp

```diff
@@ -4253,20 +4288,23 @@ bool Scope::isVariableDeclaration(const Token* const tok, const Token*& vartok,
     if (Token::Match(localVarTok, "%name% ;|=") || (localVarTok && localVarTok->varId() && localVarTok->strAt(1) == ":")) {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
     } else if (Token::Match(localVarTok, "%name% )|[") && localVarTok->str() != "operator") {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
     } else if (localVarTok && localVarTok->varId() && Token::Match(localVarTok, "%name% (|{") &&
                Token::Match(localVarTok->next()->link(), ")|} ;")) {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
     } else if (type == eCatch &&
                Token::Match(localVarTok, "%name% )")) {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
     }
-
-    return nullptr != vartok;
+    return false;
```

Comment:
I changed this because I discovered that only these four branches actually assign to vartok and as such it's possible to reach this return without ever having assigned it and thus whether the method returns true or false would depend on the variable passed in. This seemed like a recipe for bugs so I fixed it to make my tests easier to write but didn't end up using those tests in the end. I still think it's a worthwhile change.